src/internal/pfsdb/commit_provenance.go

```diff
@@ -166,3 +166,123 @@ func addCommitProvenance(tx *pachsql.Tx, from, to int) error {
 	_, err := tx.Exec(stmt, from, to)
 	return errors.Wrapf(err, "add commit provenance")
 }
+
+type commitWithDepth struct {
```

Comment:
Why does the depth need to be returned?

Reply:
I broke the query results into `commitWithDepth` and `branchWithDepth` structs to make stepping through this code with a debugger easier. I can remove these structs but I'd argue they help a lot with testing.